### ISSUE TITLE
How do I know that the transaction I submitted by the asynchronous interface is completed?

### ISSUE THREAD

**Issue by xuhuigithub** [bigchaindb/bigchaindb#2292]
I can't submit any transactions to bigchaindb, whether synchronous or asynchronous, when I deal with a large volume of transactions with the asynchronous interface . Bigchaindb will be wrong as follows(I guess it's because the network load is too high).

Bigchaindb's error when the tendermint deals with the transaction
```bash
[2018-05-15 08:36:45] [CRITICAL] (gunicorn.error) WORKER TIMEOUT (pid:1484) (webapi     - pid: 20)
```
So I want to know if I can query the status of transaction processing through bigchaindb?

**Comment by vrde** [comment]
Hi @xuhuigithub, you can use the [Stream API](https://docs.bigchaindb.com/projects/server/en/v2.0.0a5/events/websocket-event-stream-api.html). We also have an [acceptance test you can use as an example](https://github.com/bigchaindb/bigchaindb/blob/master/acceptance/python/src/test_stream.py).

**Comment by xuhuigithub** [comment]
I'm sorry I forgot to attach my version.My version of bigchaindb is 2.0.0a5.

I first submitted 1000 requests with my script using the async interface, and I sent 1000 requests with the async interface again while tendermint handled the request, and then I got the following error.

```python
Traceback (most recent call last):
  File "/home/xuhui/DEV/tests/lib/python3.6/site-packages/urllib3/connectionpool.py", line 601, in urlopen
    chunked=chunked)
  File "/home/xuhui/DEV/tests/lib/python3.6/site-packages/urllib3/connectionpool.py", line 387, in _make_request
    six.raise_from(e, None)
  File "<string>", line 2, in raise_from
  File "/home/xuhui/DEV/tests/lib/python3.6/site-packages/urllib3/connectionpool.py", line 383, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/lib/python3.6/http/client.py", line 1331, in getresponse
    response.begin()
  File "/usr/lib/python3.6/http/client.py", line 297, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python3.6/http/client.py", line 266, in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
http.client.RemoteDisconnected: Remote end closed connection without response

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/xuhui/DEV/tests/lib/python3.6/site-packages/requests/adapters.py", line 440, in send
    timeout=timeout
  File "/home/xuhui/DEV/tests/lib/python3.6/site-packages/urllib3/connectionpool.py", line 639, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "/home/xuhui/DEV/tests/lib/python3.6/site-packages/urllib3/util/retry.py", line 357, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/home/xuhui/DEV/tests/lib/python3.6/site-packages/urllib3/packages/six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "/home/xuhui/DEV/tests/lib/python3.6/site-packages/urllib3/connectionpool.py", line 601, in urlopen
    chunked=chunked)
  File "/home/xuhui/DEV/tests/lib/python3.6/site-packages/urllib3/connectionpool.py", line 387, in _make_request
    six.raise_from(e, None)
  File "<string>", line 2, in raise_from
  File "/home/xuhui/DEV/tests/lib/python3.6/site-packages/urllib3/connectionpool.py", line 383, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/lib/python3.6/http/client.py", line 1331, in getresponse
    response.begin()
  File "/usr/lib/python3.6/http/client.py", line 297, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python3.6/http/client.py", line 266, in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/lib/python3.6/unittest/case.py", line 605, in run
    testMethod()
  File "/home/xuhui/tests/BigchiaindbTests/tests.py", line 45, in test_create
    create_assest(**create_kwargs)
  File "/home/xuhui/tests/BigchiaindbTests/__init__.py", line 32, in create_assest
    sent_creation_tx = bdb.transactions.send(fulfilled_creation_tx, mode=mode)
  File "/home/xuhui/DEV/tests/lib/python3.6/site-packages/bigchaindb_driver/driver.py", line 332, in send
    headers=headers)
  File "/home/xuhui/DEV/tests/lib/python3.6/site-packages/bigchaindb_driver/transport.py", line 58, in forward_request
    headers=headers,
  File "/home/xuhui/DEV/tests/lib/python3.6/site-packages/bigchaindb_driver/connection.py", line 48, in request
    **kwargs
  File "/home/xuhui/DEV/tests/lib/python3.6/site-packages/requests/sessions.py", line 508, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/xuhui/DEV/tests/lib/python3.6/site-packages/requests/sessions.py", line 618, in send
    r = adapter.send(request, **kwargs)
  File "/home/xuhui/DEV/tests/lib/python3.6/site-packages/requests/adapters.py", line 490, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))
```

**I submit an asynchronous transaction request in parallel.**

**Comment by ttmc** [comment]
The question in the original title of this issue, "How do I know that the transaction I submitted by the asynchronous interface is completed?" was answered by @vrde in his first comment, but there's another issue that was raised, so I'll change the title to reflect _that_ issue. I'm not sure if it's still an issue in the latest release; we'll have to check.
